File: typescript/src/api/types/AgentImportSourceRefRequest.ts

export interface AgentImportSourceRefRequest {
    contentDigest: string;
    exportPolicy?: AgentImportSourceRefRequest.ExportPolicy | undefined;
    metadata?: Record<string, unknown> | undefined;
    role: string;
    storageRef: string;
}

export namespace AgentImportSourceRefRequest {
    export const ExportPolicy = {
            CustomerSourceExportable: "customer_source_exportable",
            CustomerSourcePrivate: "customer_source_private",
            ReferenceOnly: "reference_only"
        } as const;
    export type ExportPolicy = typeof ExportPolicy[keyof typeof ExportPolicy];
}

File: ruby/lib/Sikaru/types/agent_import_source_ref_request.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class AgentImportSourceRefRequest < Internal::Types::Model
      field :content_digest, -> { String }, optional: false, nullable: false, api_name: "contentDigest"

      field :export_policy, -> { Sikaru::Types::AgentImportSourceRefRequestExportPolicy }, optional: true, nullable: false, api_name: "exportPolicy"

      field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false

      field :role, -> { String }, optional: false, nullable: false

      field :storage_ref, -> { String }, optional: false, nullable: false, api_name: "storageRef"
    end
  end
end
